src/lib/sql/sql_translator.cpp

```diff
@@ -355,11 +356,21 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_join(const hsql::Join
   auto left_node = _translate_table_ref(*join.left);
   auto right_node = _translate_table_ref(*join.right);
 
-  const hsql::Expr& condition = *join.condition;
+  std::vector<const hsql::Expr*> condition_list;
+  _flatten_conjunctive_expression_tree(join.condition, condition_list);
+
+  auto get_side =
+      std::bind(&SQLTranslator::_get_side, this, std::cref(left_node), std::cref(right_node), std::placeholders::_1);
```

Comment:
This binding makes it unnecessarily hard to understand. Having `_get_side` and `get_side` doesn't help. This only prevents you from passing `left_node` and `right_node` as a parameter to `SQLTranslator::_get_sides_from_operator_expressions`, correct? I would prefer that approach, because it doesn't require people to understand what gets bound where.

Reply:
I believe I cannot convince you (for a bunch of other reasons such as usage of `std::bind`), but I really liked how `_get_sides_from_operator_expressions`worked on the very same level of abstraction, namely only `hsql::Expr`, without the need to be concerned with children LQP nodes or to directly hard-wire the decision on how to determine the sides.
I see your point (simplicity, ease of understanding) and will refactor accordingly.